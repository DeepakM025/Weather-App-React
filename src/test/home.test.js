import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import Home from "../pages/home";
import Error from "../components/error";
import WeatherDay from "../components/weatherDay";

const renderApp = () => render(<Home/>);

test("Get weather report empty input", () => {
  const {
    queryByTestId
} = renderApp();
  expect(queryByTestId('city-input').nodeValue).toBeNull;
  fireEvent.click(queryByTestId('get-weather-report'));
  expect(Error);
})

test("Get weather report", () => {
  const {
    queryByTestId
} = renderApp();
  fireEvent.change(queryByTestId('city-input'), { target: { value: 'Bengaluru' } })
  fireEvent.click(queryByTestId('get-weather-report'));
  expect(WeatherDay)
})

test("Get weather report unknown input", () => {
  const {
    queryByTestId
} = renderApp();
  fireEvent.change(queryByTestId('city-input'), { target: { value: 'dsa' } });
  fireEvent.click(queryByTestId('get-weather-report'));
  expect(Error);
})