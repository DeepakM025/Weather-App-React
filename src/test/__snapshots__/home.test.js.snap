// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get weather report empty input 1`] = `
<div
  className="container mainBody text-center"
>
  <h2>
    Weather Report
  </h2>
  <form>
    <input
      data-testid="city-input"
      onChange={[Function]}
      placeholder="Enter City..."
      type="search"
    />
    <button
      data-testid="get-weather-report"
      onClick={[Function]}
    >
      Get Weather Report
    </button>
  </form>
  <div
    className="daysWeatherWrapper"
  >
    
  </div>
</div>
`;
